src/test/kotlin/de/bund/digitalservice/useid/eidservice/EidServiceTest.kt

```diff
@@ -13,13 +17,32 @@ import org.springframework.context.annotation.Import
 @Tag("test")
 @Import(EidServiceProperties::class)
 class EidServiceTest {
+
     @Autowired
-    private lateinit var eidService: EidService
+    private lateinit var config: EidServiceConfiguration
 
     @Test
     fun `getWebserviceRequest returns a valid request`() {
+        val eidService = EidService(config)
         val webserviceRequest = eidService.webserviceRequest
-
         assertThat(webserviceRequest, isA(UseIDRequestType::class.java))
     }
+
+    @Test
+    fun `getWebserviceRequest includes specified data groups as part of operations in request`() {
+        val eidService = EidService(config, listOf("DG1", "DG4", "DG10"))
+        val operations = eidService.webserviceRequest.useOperations
+        val required = AttributeRequestType.REQUIRED
+        assertEquals(operations.documentType, required)
+        assertEquals(operations.givenNames, required)
+        assertEquals(operations.nationality, required)
+    }
+
+    @Test
+    fun `getWebserviceRequest throws IllegalStateException when passed invalid list items as data groups`() {
+        val eidService = EidService(config, listOf("iAmInvalid", "meToo"))
+        Assertions.assertThrows(IllegalStateException::class.java) {
+            eidService.webserviceRequest.useOperations
+        }
```

Comment:
do we have a case where we use forbidden data groups? let's say GST uses DG1 and DG2 but actually they mistakenly use DG4 and DG5?

Reply:
This exception would only be thrown after the setup, since the data groups will not change per request. 
They will be statically fetched from the env vars.